packages/core/src/util/token.ts

```diff
@@ -56,7 +56,12 @@ export async function getWebidFromTokenPayload(
   }
 
   if (typeof payload.webid === "string") {
-    return payload.webid;
+    if (typeof payload.azp === "string") {
+      return { webId: payload.webid, clientId: payload.azp };
+    }
+    throw new Error(
+      `The token ${JSON.stringify(payload)} is invalid: it has no 'azp' claim.`,
+    );
```

Comment:
Let's not nest this conditional here. Instead, let's define a `webid` and `clientId` variables, each initialized in their own blocks. This means `return payload.webid;` just above would become `webId = payload.webid`, and there would be a separate branch as follows: 
```javascript
if (typeof payload.azp === "string") {
  clientId = payload.azp;
}
```